src/org/mozilla/javascript/MemberBox.java

```diff
@@ -130,6 +132,68 @@ public String toString()
         return memberObject.toString();
     }
 
+    /**
+     * Function returned by calls to __lookupGetter__
+     */
+    Function asGetterFunction(final String name, final Scriptable scope, final Scriptable prototype) {
```

Comment:
Would you consider naming "scope" and "prototype" here something else, even "s" and "p", to make it clear that they are unused (at least I assume that they are, and if not, then I don't think that the lazy construction makes sense).

Reply:
Good point, maybe there is a better way to deal with this - will have a look (code was taken from the htmlunit fork of rhino written years ago).

Will write some more tests to make sure we have the right scope/protoype for the function object.